docs/specs.md

```diff
@@ -87,181 +113,323 @@ can be considered as _a collection of text chunks_. How those chunks are
+   refer to other chunks or to the full document (depending on its structure).
+
+The following sections detail these iteration interfaces for each document type.
+
+
+### Sequence documents
+
+#### Structural iteration
+
+A sequence document is divided into independent, adjacent chunks. Each chunk
+could be conceptually considered as one document paragraph (though that split is
+not strictly defined; it might be possible for a document to contain chunks
+spanning more than one typographical paragraph, or to split a very long
+paragraph into more than one chunk).
+
+Structural iteration for these documents is simple a linear sequence of
+_chunks_. Each chunk is a dictionary containing up to three elements:
+
+* **id**: an arbitrary string that should be unique per document. Its mission
+  is to make it easier later on to map detected PII instances to the chunk
+  they are part of
+* **data**: a text section that contains the textual contents of the chunk. It
+  will be a string containing UTF-8 raw text. It can contain newlines, blank
+  lines or spacing (to be considered as part of the text structure), but no
+  formatting or layout contents (it is assumed that exact formatting & layout
+  is lost when creating the source document for PII processing – this is a
+  price we pay for simplicity)
+* **context**: an optional element that, when present, contains a dictionary
+  of context fields specific for the chunk, intended to provide additional
+  data to help in the detection and decision process. Note that the context is
+  not formally part of the chunk, and the possible fields present are not
+  specified (they depend on the document original format and on the capabilities
+  of the software that extracts it).
+
+In its simplest form, the content of one chunk is, as mentioned, a document
+paragraph. But exactly how those paragraphs are determined is
+application-dependent. And in any case joining together the data elements
+in all chunks should recover a text representation of the original document.
+
+
+#### Full iteration
+
+The full iteration for sequence documents is quite similar to the structural
+iteration (given that the document structure is flat). It produces a series of
+chunks, each one of them having up to three elements:
+
+* **id**: a unique string for the chunk (there is no guarantee that is the same
+  as the id in structural iteration)
+* **data**: the same contents as in the **data** field in structural iteration
+* **context**: an optional element that, when present, contains a dictionary
+  with the same fields as in structural iteration, **plus** up to three types
+  of context structure, all of them optional:
+   - _global_: it will be a link/transposition of the information in the
+     document header, as described [above](#document-header)
+   - _before_: the data content from the chunk preceding this one
+   - _after_: the data contents from the chunk following this one
+
```

Comment:
See comment above. Before and after definately works. But there could be left and right as well. In anycase, linked data would be hard to caputre here and might just be supported by a user defined context jsonl blob that the user can pass along and use as a callback? Say, a json blob could have a {'version':0.1, 'type':, 'my content type', 'date': ..., 'content1': ...}.  Maybe not for this version of the specs. Just thinking out loud.

Reply:
Future more specific document types might have _additional_ context elements, apart from the "simple" before/after chunks. In fact the table document has one, the "column" context, which kind of a "hypertexty jump" since it points to the column name (in the header row)

We can deal with this special stuff in future versions